tools/@aws-cdk/spec2cdk/lib/cdk/resource-decider.ts

```diff
@@ -37,12 +38,47 @@ export class ResourceDecider {
 
     this.convertProperties();
     this.convertAttributes();
+    this.convertPrimaryIdentifier(); // must be called after convertProperties and convertAttributes
 
     this.propsProperties.sort((p1, p2) => p1.propertySpec.name.localeCompare(p2.propertySpec.name));
     this.classProperties.sort((p1, p2) => p1.propertySpec.name.localeCompare(p2.propertySpec.name));
     this.classAttributeProperties.sort((p1, p2) => p1.propertySpec.name.localeCompare(p2.propertySpec.name));
   }
 
+  private convertPrimaryIdentifier() {
+    for (const cfnName of this.resource.primaryIdentifier ?? []) {
+      const att = this.findAttributeByName(attributePropertyName(cfnName));
+      const prop = this.findPropertyByName(propertyNameFromCloudFormation(cfnName));
+      if (att) {
+        this.primaryIdentifier.push(att);
+      } else if (prop) {
+        // rename the prop name as an attribute name, since it is gettable by ref
+        this.primaryIdentifier.push({
+          ...prop,
+          name: attributePropertyName(prop.name[0].toUpperCase() + prop.name.slice(1)),
+          docs: {
+            ...prop.docs,
+            remarks: prop.docs?.remarks?.concat(['\n', `@cloudformationRef ${prop.name}`].join('\n')),
```

Comment:
not sure what remark we want to add here... does `@cloudformationRef` make sense? for attributes, its `@cloudformationAttribute`. And do we want to do the same for primaryIdentifiers found in attributes? I think yes, but wanna check

Reply:
Pretty sure `@cloudformationAttribute` is functionally used during linting. So we might not need anything here.